lib/matplotlib/figure.py

```diff
@@ -2614,12 +2614,16 @@ def show(self, warn=True):
             may only be shown briefly or not shown at all if you or your
             environment are not managing an event loop.
 
-            Proper use cases for `.Figure.show` include running this from a
-            GUI application or an IPython shell.
-
-            If you're running a pure python shell or executing a non-GUI
-            python script, you should use `matplotlib.pyplot.show` instead,
-            which takes care of managing the event loop for you.
+            Proper use cases for `.Figure.show` include running this from a GUI
+            application (where there is persistently an event loop running) or
+            from a shell with an input hook configured (IPython will do this).
+            Some, but not all, GUI toolkits will register an input hook on
+            import.  See :ref:`cp_integration` for more details.
```

Comment:
I think being specific helps:

```suggestion
            Use cases for `.Figure.show` include running this from a GUI
            application (where there is persistently an event loop running) or
            from a shell, like IPython, that install an input hook to allow the interactive 
            shell to accept input while the figure is also being shown and interactive.
            Some, but not all, GUI toolkits will register an input hook on
            import.  See :ref:`cp_integration` for more details.
```

However, I'm still not sure what you mean by a "GUI toolkit" in this context.  Do you mean our backends?  Or do you mean Qt etc install an input hook on import, even in the absence of Matplotlib?

Reply:
Thank you, that is much better!